scene/property_utils.cpp

```diff
@@ -89,6 +89,16 @@ Variant PropertyUtils::get_property_default_value(const Object *p_object, const
 		*r_is_valid = false;
 	}
 
+	// Handle special case "script" property, where the default value is either null or the custom type script.
+	// Do this only if there's no states stack cache to trace for default values.
+	if (p_property == CoreStringName(script) && p_object->has_meta(SceneStringName(__custom_type_script)) && !p_states_stack_cache) {
```

Comment:
```suggestion
	if (p_property == CoreStringName(script) && !p_states_stack_cache && p_object->has_meta(SceneStringName(__custom_type_script))) {
```
Cheaper check first.